tests/scm/scm-test-arity.scm

```diff
@@ -0,0 +1,43 @@
+
+(use-modules (opencog) (opencog test-runner))
+(use-modules (opencog python))
+
+(opencog-test-runner)
+(define t "opencog-arity-test")
+
+(test-begin t)
+
+(python-eval "
+from opencog.atomspace import AtomSpace, TruthValue
+from opencog.atomspace import types
+def foo(atspace):
+    TV = TruthValue(0.2, 0.69)
+    atspace.add_node(types.ConceptNode, 'Apple', TV)
+")
+
+(python-call-with-as "foo" (cog-atomspace))
+(cog-node 'ConceptNode "Apple")
+
+; Return the strength of a simple truth value
+(define (get-tv-strength tv) (cdr (assoc 'mean (cog-tv->alist tv))))
+
+(define strength (get-tv-strength
+ (cog-tv (cog-node 'ConceptNode "Apple"))))
+
+(test-assert "python-eval is borken" (< (- 0.2 strength) 0.00001))
+
+
+(define (catch-wrong-args thunk)
+  (catch #t 
+    thunk
+    (lambda (key . parameters)
+      (format (current-error-port)
+              "caught throw to '~a: ~a\n" key parameters)
+      "catch")))
+
+(define failed-result
+  (catch-wrong-args
+   (lambda () (python-call-with-as "foo" (cog-atomspace) (cog-node 'ConceptNode "Test")))))
```

Comment:
I don't think this does what you think it does. Because you said `cog-node` instead of `cog-new-node`, the `(Concept "test")` was never created. Instead, you are passing `#f` to "foo".  I'm not sure what happens if you pass `#f` instead of passing an actual Atom. Maybe some other exception gets thrown, but not the wrong-arguments exception??

Reply:
This remains unfixed.